aubio recipe created

jithesh82 opened the pull request:
Hello there,

Thanks a lot for you wonderful work.  I have created a recipe for aubio package.  I have tested it on my android phone using buildozer.  It works well: for example able to do pitch detection.  

Please let me know if you need any further information. 

Thanks again,
-Jithesh

jithesh82 committed:
Create test.txt

creating aubio folder

jithesh82 committed:
uploading aubio recipe

jithesh82 committed:
Delete pythonforandroid/recipes/aubio/test.txt

jithesh82 committed:
Update __init__.py

cleaned up - removed unwanted comments

AndreMiras reviewed (commented):
Thanks Jithesh, this is looking good.
Would you mind addressing the linting issues?
https://github.com/kivy/python-for-android/actions/runs/10172437349/job/28144046505
```
pythonforandroid/recipes/aubio/__init__.py:2:1: F401 'pythonforandroid.recipe.CompiledComponentsPythonRecipe' imported but unused
pythonforandroid/recipes/aubio/__init__.py:10:14: E225 missing whitespace around operator
pythonforandroid/recipes/aubio/__init__.py:14:1: W391 blank line at end of file
```

jithesh82 committed:
Update __init__.py

removed lines at the end of the file.  removed os import - unused

jithesh82 committed:
Create temp.txt

test

jithesh82 committed:
Delete pythonforandroid/recipes/aubio/__init__.py

jithesh82 committed:
Add files via upload

reformatted for linting

jithesh82 committed:
Delete pythonforandroid/recipes/aubio/temp.txt

jithesh82 commented:
Hello @AndreMiras ,

Thanks a lot for your prompt response.  I have removed two blank lines at the end.  I have reformatted the code using black.  Also tried ruff.  Please let me know if it is okay.

jithesh82 committed:
Create test.txt

jithesh82 committed:
Delete pythonforandroid/recipes/aubio/__init__.py

jithesh82 committed:
Add files via upload

updated the import line as directed by flake8

jithesh82 committed:
Delete pythonforandroid/recipes/aubio/test.txt

jithesh82 commented:
Hello @AndreMiras ,

I have gone through the following errors and fixed them:

1)  pythonforandroid/recipes/aubio/__init__.py:2:1: F401 'pythonforandroid.recipe.CompiledComponentsPythonRecipe' imported but unused
Removed the unused import.

2) pythonforandroid/recipes/aubio/__init__.py:4:1: E302 expected 2 blank lines, found 1
Added blank lines after import

3) pythonforandroid/recipes/aubio/__init__.py:8:14: E225 missing whitespace around operator
Added white spaces around operators:  ' = '

4) pythonforandroid/recipes/aubio/__init__.py:8:15: F821 undefined name 'join'
This one was accidentally removed when I was cleaning.  I have put it back.

5) pythonforandroid/recipes/aubio/__init__.py:10:1: E305 expected 2 blank lines after class or function definition, found 1
Added two blank lines after class definition.

I believe it passes the test now.  Please let me know.  

Thank you!
-Jithesh

AndreMiras commented:
Yes I was looking this up, but the "Test updated recipes" is failing meaning that the recipe doesn't compile for some reason.
However impossible to download the full log so far

jithesh82 commented:
Thank you so much.  I am going to compile it on my computer using buildozer
and let you know.

On Wed, Jul 31, 2024 at 2:10 PM Andre Miras ***@***.***>
wrote:

> Yes I was looking this up, but the "Test updated recipes" is failing
> meaning that the recipe doesn't compile for some reason.
> However impossible to download the full log so far
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2261108273>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLR5ZES5KELSQZRYZQYDZPESCJAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDENRRGEYDQMRXGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

AndreMiras commented:
I could download the logs eventually, they're very hard to read in the CI, we need to address that.
This is what I extracted and found relevant. I'm not 100% they're critical errors for the build or false errors.
```
...<truncated>...
2024-07-31T16:12:39.3193768Z     raise exc
2024-07-31T16:12:39.3193933Z 
2024-07-31T16:12:39.3194079Z sh.ErrorReturnCode_1: 
2024-07-31T16:12:39.3194086Z 
2024-07-31T16:12:39.3194092Z 
2024-07-31T16:12:39.3194097Z 
2024-07-31T16:12:39.3195403Z   RAN: /home/user/app/venv/bin/python3 '-cimport sys; from Cython.Compiler.Main import setuptools_main; sys.exit(setuptools_main());' ./.eggs/numpy-2.0.1-py3.11-linux-x86_64.egg/numpy/random/_examples/cython/extending_distributions.pyx
2024-07-31T16:12:39.3195421Z 
2024-07-31T16:12:39.3195427Z 
2024-07-31T16:12:39.3195431Z 
2024-07-31T16:12:39.3195552Z   STDOUT:
2024-07-31T16:12:39.3195558Z 
2024-07-31T16:12:39.3195563Z 
2024-07-31T16:12:39.3195569Z 
2024-07-31T16:12:39.3195717Z Error compiling Cython file:
2024-07-31T16:12:39.3195723Z 
2024-07-31T16:12:39.3195904Z ------------------------------------------------------------
2024-07-31T16:12:39.3195911Z 
2024-07-31T16:12:39.3196015Z ...
2024-07-31T16:12:39.3196023Z 
2024-07-31T16:12:39.3196298Z import numpy as np
2024-07-31T16:12:39.3196307Z 
2024-07-31T16:12:39.3196446Z cimport numpy as np
2024-07-31T16:12:39.3196454Z 
2024-07-31T16:12:39.3196572Z cimport cython
2024-07-31T16:12:39.3196579Z 
2024-07-31T16:12:39.3196933Z from cpython.pycapsule cimport PyCapsule_IsValid, PyCapsule_GetPointer
2024-07-31T16:12:39.3196942Z 
2024-07-31T16:12:39.3197103Z from libc.stdint cimport uint16_t, uint64_t
2024-07-31T16:12:39.3197238Z 
2024-07-31T16:12:39.3197396Z from numpy.random cimport bitgen_t
2024-07-31T16:12:39.3197402Z 
2024-07-31T16:12:39.3197507Z ^
2024-07-31T16:12:39.3197512Z 
2024-07-31T16:12:39.3197688Z ------------------------------------------------------------
2024-07-31T16:12:39.3197695Z 
2024-07-31T16:12:39.3197701Z 
2024-07-31T16:12:39.3197713Z 
2024-07-31T16:12:39.3198437Z .eggs/numpy-2.0.1-py3.11-linux-x86_64.egg/numpy/random/_examples/cython/extending_distributions.pyx:11:0: 'numpy/random.pxd' not found
2024-07-31T16:12:39.3198447Z 
2024-07-31T16:12:39.3198453Z 
2024-07-31T16:12:39.3198463Z 
2024-07-31T16:12:39.3198611Z Error compiling Cython file:
2024-07-31T16:12:39.3198621Z 
2024-07-31T16:12:39.3198810Z ------------------------------------------------------------
2024-07-31T16:12:39.3198817Z 
2024-07-31T16:12:39.3198918Z ...
2024-07-31T16:12:39.3198924Z 
2024-07-31T16:12:39.3199235Z import numpy as np
2024-07-31T16:12:39.3199245Z 
2024-07-31T16:12:39.3199386Z cimport numpy as np
2024-07-31T16:12:39.3199392Z 
2024-07-31T16:12:39.3199509Z cimport cython
2024-07-31T16:12:39.3199516Z 
2024-07-31T16:12:39.3199906Z from cpython.pycapsule cimport PyCapsule_IsValid, PyCapsule_GetPointer
2024-07-31T16:12:39.3199914Z 
...<truncated>...
2024-07-31T16:12:39.3278173Z 
2024-07-31T16:12:39.3278183Z 
2024-07-31T16:12:39.3279038Z .eggs/numpy-2.0.1-py3.11-linux-x86_64.egg/numpy/random/_examples/cython/extending_distributions.pyx:13:0: 'numpy/random/c_distributions.pxd' not found
2024-07-31T16:12:39.3279059Z 
2024-07-31T16:12:39.3279064Z 
2024-07-31T16:12:39.3279069Z 
2024-07-31T16:12:39.3279229Z Error compiling Cython file:
2024-07-31T16:12:39.3279236Z 
2024-07-31T16:12:39.3279418Z ------------------------------------------------------------
2024-07-31T16:12:39.3279426Z 
2024-07-31T16:12:39.3279535Z ...
2024-07-31T16:12:39.3279543Z 
2024-07-31T16:12:39.3279661Z cimport cython
2024-07-31T16:12:39.3279668Z 
2024-07-31T16:12:39.3280017Z from cpython.pycapsule cimport PyCapsule_IsValid, PyCapsule_GetPointer
...<truncated>...
2024-07-31T16:12:39.3336678Z from libc.stdint cimport uint16_t, uint64_t
2024-07-31T16:12:39.3336695Z 
2024-07-31T16:12:39.3336849Z from numpy.random cimport bitgen_t
2024-07-31T16:12:39.3336855Z 
2024-07-31T16:12:39.3337001Z from numpy.random import PCG64
2024-07-31T16:12:39.3337007Z 
2024-07-31T16:12:39.3337184Z from numpy.random.c_distributions cimport (
2024-07-31T16:12:39.3337198Z 
2024-07-31T16:12:39.3337302Z ^
2024-07-31T16:12:39.3337308Z 
2024-07-31T16:12:39.3337486Z ------------------------------------------------------------
2024-07-31T16:12:39.3337493Z 
2024-07-31T16:12:39.3337498Z 
2024-07-31T16:12:39.3337510Z 
2024-07-31T16:12:39.3338852Z .eggs/numpy-2.0.1-py3.11-linux-x86_64.egg/numpy/random/_examples/cython/extending_distributions.pyx:13:0: 'numpy/random/c_distributions/random_standard_uniform_fill.pxd' not found
2024-07-31T16:12:39.3338872Z 
2024-07-31T16:12:39.3338877Z 
2024-07-31T16:12:39.3338881Z 
2024-07-31T16:12:39.3339035Z Error compiling Cython file:
2024-07-31T16:12:39.3339042Z 
2024-07-31T16:12:39.3339233Z ------------------------------------------------------------
2024-07-31T16:12:39.3339240Z 
2024-07-31T16:12:39.3339342Z ...
2024-07-31T16:12:39.3339349Z 
2024-07-31T16:12:39.3339473Z cimport cython
2024-07-31T16:12:39.3339479Z 
2024-07-31T16:12:39.3339824Z from cpython.pycapsule cimport PyCapsule_IsValid, PyCapsule_GetPointer
2024-07-31T16:12:39.3339839Z 
2024-07-31T16:12:39.3340006Z from libc.stdint cimport uint16_t, uint64_t
2024-07-31T16:12:39.3340017Z 
2024-07-31T16:12:39.3340175Z from numpy.random cimport bitgen_t
2024-07-31T16:12:39.3340182Z 
2024-07-31T16:12:39.3340325Z from numpy.random import PCG64
2024-07-31T16:12:39.3340332Z 
2024-07-31T16:12:39.3340511Z from numpy.random.c_distributions cimport (
2024-07-31T16:12:39.3340528Z 
2024-07-31T16:12:39.3340632Z ^
2024-07-31T16:12:39.3340638Z 
...<truncated>...
```
If you want to give it a try, I downloaded the logs with:
```sh
wget -c 'https://productionresultssa4.blob.core.windows.net/actions-results/e47428b8-4113-4f02-9c27-35501a3baca1/workflow-job-run-27bf7f3f-2775-5cff-c17f-312f58940964/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-07-31T18%3A21%3A25Z&sig=c%2Fa14t9Pg7XLGDXaohKv56VUyyngb%2FqMflnioFueoIE%3D&ske=2024-08-01T04%3A55%3A44Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-07-31T16%3A55%3A44Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-05-04&sp=r&spr=https&sr=b&st=2024-07-31T18%3A11%3A20Z&sv=2024-05-04'
```

jithesh82 commented:
Thank you!  It gives me an error:  ERROR 403: Server failed to authenticate
the request. Make sure the value of the Authorization header is formed
correctly including the signature..

On Wed, Jul 31, 2024 at 2:28 PM Andre Miras ***@***.***>
wrote:

> I could download the logs eventually, they're very hard to read in the CI,
> we need to address that.
> This is what I extracted and found relevant. I'm not 100% they're critical
> errors for the build or false errors.
>
> ...<truncated>...
> 2024-07-31T16:12:39.3193768Z     raise exc
> 2024-07-31T16:12:39.3193933Z
> 2024-07-31T16:12:39.3194079Z sh.ErrorReturnCode_1:
> 2024-07-31T16:12:39.3194086Z
> 2024-07-31T16:12:39.3194092Z
> 2024-07-31T16:12:39.3194097Z
> 2024-07-31T16:12:39.3195403Z   RAN: /home/user/app/venv/bin/python3 '-cimport sys; from Cython.Compiler.Main import setuptools_main; sys.exit(setuptools_main());' ./.eggs/numpy-2.0.1-py3.11-linux-x86_64.egg/numpy/random/_examples/cython/extending_distributions.pyx
> 2024-07-31T16:12:39.3195421Z
> 2024-07-31T16:12:39.3195427Z
> 2024-07-31T16:12:39.3195431Z
> 2024-07-31T16:12:39.3195552Z   STDOUT:
> 2024-07-31T16:12:39.3195558Z
> 2024-07-31T16:12:39.3195563Z
> 2024-07-31T16:12:39.3195569Z
> 2024-07-31T16:12:39.3195717Z Error compiling Cython file:
> 2024-07-31T16:12:39.3195723Z
> 2024-07-31T16:12:39.3195904Z ------------------------------------------------------------
> 2024-07-31T16:12:39.3195911Z
> 2024-07-31T16:12:39.3196015Z ...
> 2024-07-31T16:12:39.3196023Z
> 2024-07-31T16:12:39.3196298Z import numpy as np
> 2024-07-31T16:12:39.3196307Z
> 2024-07-31T16:12:39.3196446Z cimport numpy as np
> 2024-07-31T16:12:39.3196454Z
> 2024-07-31T16:12:39.3196572Z cimport cython
> 2024-07-31T16:12:39.3196579Z
> 2024-07-31T16:12:39.3196933Z from cpython.pycapsule cimport PyCapsule_IsValid, PyCapsule_GetPointer
> 2024-07-31T16:12:39.3196942Z
> 2024-07-31T16:12:39.3197103Z from libc.stdint cimport uint16_t, uint64_t
> 2024-07-31T16:12:39.3197238Z
> 2024-07-31T16:12:39.3197396Z from numpy.random cimport bitgen_t
> 2024-07-31T16:12:39.3197402Z
> 2024-07-31T16:12:39.3197507Z ^
> 2024-07-31T16:12:39.3197512Z
> 2024-07-31T16:12:39.3197688Z ------------------------------------------------------------
> 2024-07-31T16:12:39.3197695Z
> 2024-07-31T16:12:39.3197701Z
> 2024-07-31T16:12:39.3197713Z
> 2024-07-31T16:12:39.3198437Z .eggs/numpy-2.0.1-py3.11-linux-x86_64.egg/numpy/random/_examples/cython/extending_distributions.pyx:11:0: 'numpy/random.pxd' not found
> 2024-07-31T16:12:39.3198447Z
> 2024-07-31T16:12:39.3198453Z
> 2024-07-31T16:12:39.3198463Z
> 2024-07-31T16:12:39.3198611Z Error compiling Cython file:
> 2024-07-31T16:12:39.3198621Z
> 2024-07-31T16:12:39.3198810Z ------------------------------------------------------------
> 2024-07-31T16:12:39.3198817Z
> 2024-07-31T16:12:39.3198918Z ...
> 2024-07-31T16:12:39.3198924Z
> 2024-07-31T16:12:39.3199235Z import numpy as np
> 2024-07-31T16:12:39.3199245Z
> 2024-07-31T16:12:39.3199386Z cimport numpy as np
> 2024-07-31T16:12:39.3199392Z
> 2024-07-31T16:12:39.3199509Z cimport cython
> 2024-07-31T16:12:39.3199516Z
> 2024-07-31T16:12:39.3199906Z from cpython.pycapsule cimport PyCapsule_IsValid, PyCapsule_GetPointer
> 2024-07-31T16:12:39.3199914Z
> ...<truncated>...
> 2024-07-31T16:12:39.3278173Z
> 2024-07-31T16:12:39.3278183Z
> 2024-07-31T16:12:39.3279038Z .eggs/numpy-2.0.1-py3.11-linux-x86_64.egg/numpy/random/_examples/cython/extending_distributions.pyx:13:0: 'numpy/random/c_distributions.pxd' not found
> 2024-07-31T16:12:39.3279059Z
> 2024-07-31T16:12:39.3279064Z
> 2024-07-31T16:12:39.3279069Z
> 2024-07-31T16:12:39.3279229Z Error compiling Cython file:
> 2024-07-31T16:12:39.3279236Z
> 2024-07-31T16:12:39.3279418Z ------------------------------------------------------------
> 2024-07-31T16:12:39.3279426Z
> 2024-07-31T16:12:39.3279535Z ...
> 2024-07-31T16:12:39.3279543Z
> 2024-07-31T16:12:39.3279661Z cimport cython
> 2024-07-31T16:12:39.3279668Z
> 2024-07-31T16:12:39.3280017Z from cpython.pycapsule cimport PyCapsule_IsValid, PyCapsule_GetPointer
> ...<truncated>...
> 2024-07-31T16:12:39.3336678Z from libc.stdint cimport uint16_t, uint64_t
> 2024-07-31T16:12:39.3336695Z
> 2024-07-31T16:12:39.3336849Z from numpy.random cimport bitgen_t
> 2024-07-31T16:12:39.3336855Z
> 2024-07-31T16:12:39.3337001Z from numpy.random import PCG64
> 2024-07-31T16:12:39.3337007Z
> 2024-07-31T16:12:39.3337184Z from numpy.random.c_distributions cimport (
> 2024-07-31T16:12:39.3337198Z
> 2024-07-31T16:12:39.3337302Z ^
> 2024-07-31T16:12:39.3337308Z
> 2024-07-31T16:12:39.3337486Z ------------------------------------------------------------
> 2024-07-31T16:12:39.3337493Z
> 2024-07-31T16:12:39.3337498Z
> 2024-07-31T16:12:39.3337510Z
> 2024-07-31T16:12:39.3338852Z .eggs/numpy-2.0.1-py3.11-linux-x86_64.egg/numpy/random/_examples/cython/extending_distributions.pyx:13:0: 'numpy/random/c_distributions/random_standard_uniform_fill.pxd' not found
> 2024-07-31T16:12:39.3338872Z
> 2024-07-31T16:12:39.3338877Z
> 2024-07-31T16:12:39.3338881Z
> 2024-07-31T16:12:39.3339035Z Error compiling Cython file:
> 2024-07-31T16:12:39.3339042Z
> 2024-07-31T16:12:39.3339233Z ------------------------------------------------------------
> 2024-07-31T16:12:39.3339240Z
> 2024-07-31T16:12:39.3339342Z ...
> 2024-07-31T16:12:39.3339349Z
> 2024-07-31T16:12:39.3339473Z cimport cython
> 2024-07-31T16:12:39.3339479Z
> 2024-07-31T16:12:39.3339824Z from cpython.pycapsule cimport PyCapsule_IsValid, PyCapsule_GetPointer
> 2024-07-31T16:12:39.3339839Z
> 2024-07-31T16:12:39.3340006Z from libc.stdint cimport uint16_t, uint64_t
> 2024-07-31T16:12:39.3340017Z
> 2024-07-31T16:12:39.3340175Z from numpy.random cimport bitgen_t
> 2024-07-31T16:12:39.3340182Z
> 2024-07-31T16:12:39.3340325Z from numpy.random import PCG64
> 2024-07-31T16:12:39.3340332Z
> 2024-07-31T16:12:39.3340511Z from numpy.random.c_distributions cimport (
> 2024-07-31T16:12:39.3340528Z
> 2024-07-31T16:12:39.3340632Z ^
> 2024-07-31T16:12:39.3340638Z
> ...<truncated>...
>
> If you want to give it a try, I downloaded the logs with:
>
> wget -c 'https://productionresultssa4.blob.core.windows.net/actions-results/e47428b8-4113-4f02-9c27-35501a3baca1/workflow-job-run-27bf7f3f-2775-5cff-c17f-312f58940964/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-07-31T18%3A21%3A25Z&sig=c%2Fa14t9Pg7XLGDXaohKv56VUyyngb%2FqMflnioFueoIE%3D&ske=2024-08-01T04%3A55%3A44Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-07-31T16%3A55%3A44Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-05-04&sp=r&spr=https&sr=b&st=2024-07-31T18%3A11%3A20Z&sv=2024-05-04'
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2261154535>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLR7OSFN2X6LSZT4LRGDZPEUEFAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDENRRGE2TINJTGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

AndreMiras commented:
I've basically copied the link "View raw logs" from one of the failed "Test updated recipes" build.
![image](https://github.com/user-attachments/assets/f3ebe094-774e-4528-b188-e5d05503eb30)

jithesh82 commented:
Hello @AndreMiras,

On my computer I am able to build aubio without errors for all 4 abi's.  I also tested them all on emulators without error except arm64-v8a for which I need to find an emulator to test on.  I was using buildozer though, not p4a directly with arguments.  
I looked through your log once.  I'll check if I can reproduce it on my machine.  Thank you!

AndreMiras commented:
Hi @jithesh82,
I've fixed the build logs so it's easier to catch errors.
Do you mind rebasing to develop so we can more clearly see how the build is failing

jithesh82 commented:
Hello @AndreMiras  

Thank you for your message.  I rebased the fork.  I haven't done it before, I followed instructions from medium and chagpt.  I am not sure it is correct.

I had downloaded the log you pointed out and ran the rebuild-recipe code locally on my machine on docker.  It shows exact error message.  can't find a header file.  I followed the log and used the same ndk and api versions.  My buildozer still runs fine.  

Previously I had tried to compile aubio package on my machine for android following the instructions on aubio readme.  I manually downloaded compilers and I had similar issues of not being able to locate headers.  So one thing I noticed from your log, after downloading and setting up ndk and sdk, it removes the prebuilt compilers and replaces them with another one as bellow, if I understand correctly.  This is one step I didn't do while using buildozer and I could try to reproduce the error log.

Android API level is          : 27
Android NDK version is        : 25b
Android NDK Legacy version is : 21e

```
mkdir  /home/user/.android/android-ndk-legacy/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64
tar -xf gcc-arm64-linux-x86_64.tar.bz2 -C /home/user/.android/android-ndk-legacy/toolchains/aarch64-linux-android-4.9/prebuilt/linux-x86_64/ --strip-components 1
rm -rf /home/user/.android/android-ndk-legacy/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/
mkdir  /home/user/.android/android-ndk-legacy/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/
tar -xf gcc-arm-linux-x86_64.tar.bz2 -C /home/user/.android/android-ndk-legacy/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/ --strip-components 1
```
Thanks a lot for all your work and help.

AndreMiras commented:
Thanks for looking this up.
I don't think the rebase worked as expected.
Ping me on Discord and we can try to sort it out tomorrow (CEST)
Edit: no need to close and open a new PR, the commits can still be fixed

jithesh82 commented:
Yes, I'll ping you tomorrow.  Thank you!

On Mon, Aug 19, 2024 at 4:03 PM Andre Miras ***@***.***>
wrote:

> Thanks for looking this up.
> I don't think the rebase worked as expected.
> Ping me on Discord and we can try to sort it out tomorrow (CEST)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2297349763>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLR4BG62B5ITG73IWZRLZSJFSLAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEOJXGM2DSNZWGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

jithesh82 commented:
Hello Andre,

I send a message to you on discord.   It wasn't delivered because 'we don't
share servers or may your inbox only accept messages from friends'.  What
shall I do? Thank you!

On Mon, Aug 19, 2024 at 4:03 PM Andre Miras ***@***.***>
wrote:

> Thanks for looking this up.
> I don't think the rebase worked as expected.
> Ping me on Discord and we can try to sort it out tomorrow (CEST)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2297349763>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLR4BG62B5ITG73IWZRLZSJFSLAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEOJXGM2DSNZWGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

AndreMiras commented:
Thanks for the update, PR looks good.
Unfortunately the new recipe didn't build, so no way to know how the CI handles the change.
```
[INFO]:    Building updated recipes for the following archs: ['x86']
[INFO]:    recipes modified: set()
[INFO]:    recipes to build: set()
[INFO]:    recipes to build (no broken): set()
```
I suspect this is because the branch name is the same on the fork.
I'll try to see how I can fix that case in our CI later.
I'll keep you posted.

AndreMiras commented:
Fixed the CI, do you mind rebasing again?
```
git pull --rebase upstream develop
```

jithesh82 commented:
I did that:
`From github.com:jithesh82/python-for-android
 * branch              develop    -> FETCH_HEAD
Already up to date.`

Thank you so much for your prompt work.

On Tue, Aug 20, 2024 at 4:10 PM Andre Miras ***@***.***>
wrote:

> Fixed the CI, do you mind rebasing again?
>
> git pull --rebase origin develop
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2299678751>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLR4FKMKT6HUEPF5MWZLZSOPB7AVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEOJZGY3TQNZVGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

AndreMiras commented:
You need to use `upstream` not `origin` in the rebase command

jithesh82 commented:
Now, I did.  Thank you!
git pull --rebase upstream develop

`remote: Enumerating objects: 25, done.
remote: Counting objects: 100% (25/25), done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 15 (delta 9), reused 12 (delta 8), pack-reused 0 (from 0)
Unpacking objects: 100% (15/15), 3.04 KiB | 194.00 KiB/s, done.
From https://github.com/kivy/python-for-android
 * branch              develop    -> FETCH_HEAD
   3094c048..182bec59  develop    -> upstream/develop
Successfully rebased and updated refs/heads/develop.
`

On Tue, Aug 20, 2024 at 5:41 PM Andre Miras ***@***.***>
wrote:

> You need to use upstream not origin in the rebase command
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2299813327>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLRY3PTWWKKUPLZQ2MMLZSOZZLAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEOJZHAYTGMZSG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

AndreMiras commented:
Perfect, then you need to `push --force` the change :)

jithesh82 commented:
I just did.  Thank you!

`Username for 'https://github.com': jithesh82
Password for ***@***.***':
Enumerating objects: 716, done.
Counting objects: 100% (524/524), done.
Delta compression using up to 4 threads
Compressing objects: 100% (165/165), done.
Writing objects: 100% (397/397), 95.90 KiB | 13.70 MiB/s, done.
Total 397 (delta 244), reused 359 (delta 207), pack-reused 0
remote: Resolving deltas: 100% (244/244), completed with 57 local objects.
To https://github.com/jithesh82/python-for-android.git
 + 3e982e56...2e795466 develop -> develop (forced update)`

On Wed, Aug 21, 2024 at 2:30 AM Andre Miras ***@***.***>
wrote:

> Perfect, then you need to push --force the change :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2301236036>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLR2BT2EKVHBMI7C5GN3ZSQXXDAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMBRGIZTMMBTGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

AndreMiras commented on pythonforandroid/recipes/aubio/__init__.py:
```diff
@@ -0,0 +1,12 @@
+class AubioRecipe(CythonRecipe):
+    version = "0.4.7"
+    url = "https://aubio.org/pub/aubio-{version}.tar.bz2"
+    depends = ["numpy"]  # Make sure 'samplerate' is included as a dependency
```
this comment is confusing me

AndreMiras commented:
Thanks, something is off, I see the old commits went back, a rebase shouldn't have done that.
Also the clone URL changed, we updated the remote the other day to be using SSH rather than HTTPS as if it was pushed from another directory that had another clone.
No worries let's first see what the CI is giving, all we need is for it to be rebased, squashing can still be done upon merging

jithesh82 commented:
Nice.  Thank you very much!

depends = ["numpy"]  # Make sure 'samplerate' is included as a dependency

Is the above confusing?  I believe it was somewhere from the aubio package.
It was a lot of trial and error before I got it working on my machine.  We
don't need it. I could remove it.



On Wed, Aug 21, 2024 at 9:11 AM Andre Miras ***@***.***>
wrote:

> Thanks, something is off, I see the old commits went back, a rebase
> shouldn't have done that.
> Also the clone URL changed, we updated the remote the other day to be
> using SSH rather than HTTPS as if it was pushed from another directory that
> had another clone.
> No worries let's first see what the CI is giving, all we need is for it to
> be rebased, squashing can still be done upon merging
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2302024095>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLR4LBBU72Z44CRX2253ZSSGZRAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMBSGAZDIMBZGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

jithesh82 commented:
Andre, what you suspected yesterday was right regarding compilation
failure.  When I compile aubio on my local machine, where numpy is not
installed I get the similar error.

`python setup.py build

gcc -pthread -B /home/jk/miniconda3/compiler_compat -DNDEBUG -fwrapv -O2
-Wall -fPIC -O2 -isystem /home/jk/miniconda3/include -fPIC -O2 -isystem
/home/jk/miniconda3/include -fPIC -DAUBIO_VERSION=0.4.7 -DUSE_LOCAL_AUBIO=1
-DHAVE_STDLIB_H=1 -DHAVE_STDIO_H=1 -DHAVE_MATH_H=1 -DHAVE_STRING_H=1
-DHAVE_C99_VARARGS_MACROS=1 -DHAVE_LIMITS_H=1 -DHAVE_STDARG_H=1
-DHAVE_MEMCPY_HACKS=1 -DHAVE_SAMPLERATE=1 -DHAVE_WAVWRITE=1
-DHAVE_WAVREAD=1 -Ipython/ext -Isrc -Ipython/gen
-I/home/jk/miniconda3/include/python3.11 -c python/ext/aubiomodule.c -o
build/temp.linux-x86_64-cpython-311/python/ext/aubiomodule.o
In file included from python/ext/aubiomodule.c:2:
python/ext/aubio-types.h:16:10: fatal error: numpy/arrayobject.h: No such
file or directory
   16 | #include <numpy/arrayobject.h>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.`

This is a miniconda python3.11 environment.  I have another environment
where numpy is installed, where I had been building aubio using buildozer.
When I issue the same command, on that env, compilation goes smooth and
easy.

In the setup.py there is a line saying `numpy.get_include()`.  But it
wouldn't complain and proceed because, it is written as

`try:
    import numpy
    include_dirs += [ numpy.get_include() ]
except ImportError:
    pass
`
But as the compilation proceeds it can't find the headers and it fails, I
believe.

So we need host-python with numpy installed for the build to be successful,
as you rightly pointed out.  I don't know how to do that.

On Wed, Aug 21, 2024 at 9:37 AM Jithesh Kuyyalil ***@***.***>
wrote:

> Nice.  Thank you very much!
>
> depends = ["numpy"]  # Make sure 'samplerate' is included as a dependency
>
> Is the above confusing?  I believe it was somewhere from the aubio
> package. It was a lot of trial and error before I got it working on my
> machine.  We don't need it. I could remove it.
>
>
>
> On Wed, Aug 21, 2024 at 9:11 AM Andre Miras ***@***.***>
> wrote:
>
>> Thanks, something is off, I see the old commits went back, a rebase
>> shouldn't have done that.
>> Also the clone URL changed, we updated the remote the other day to be
>> using SSH rather than HTTPS as if it was pushed from another directory that
>> had another clone.
>> No worries let's first see what the CI is giving, all we need is for it
>> to be rebased, squashing can still be done upon merging
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2302024095>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AEICLR4LBBU72Z44CRX2253ZSSGZRAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMBSGAZDIMBZGU>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
>
>
> -Love Note,
>
> Dr. Jithesh Kuyyalil  (He/him/his)
>
> IBO @ Amway.
>
> ------------------------------
>
> Mobile 617-304-5563
>
> Web https://jitheshkuyyalil.com  Email ***@***.***
> Blog  https://joyoflife.blog  Amway Shop: https:/
> /www.amway.com/myshop/jitheshkuyyalil
> <https://www.amway.com/myshop/jitheshkuyyalil>
>
> *I BELIEVE IN YOU*
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

AndreMiras commented:
OK I probably misread the code comment. I thought we meant to include "samplerate" in this array, but maybe we mean that having "numpy" would include "samplerate", I'm not sure what we mean exactly here.

Also now that the "Test updated recipes" is working we see more clearly what the error was:
```
python/ext/aubio-types.h:16:10: fatal error: 'numpy/arrayobject.h' file not found
#include <numpy/arrayobject.h>
```
Check on other recipes how numpy headers are handled.
I would recommend trying locally from a docker environment to reproduce the setup we have in the CI.
```
docker build --tag=kivy/python-for-android .
time docker run -it \
  --volume $(pwd)/pythonforandroid:/home/user/app/pythonforandroid \
  --env ANDROID_NDK_HOME_LEGACY=/home/user/.android/android-ndk-legacy \
  --env ANDROID_SDK_HOME=/home/user/.android/android-sdk \
  --env ANDROID_NDK_HOME=/home/user/.android/android-ndk \
  --rm kivy/python-for-android bash
```
The volume mount is so that you can edit your recipe without always rebuilding the container.
Once you're inside the running container you can try to build your apk e.g.
```
. venv/bin/activate
cd testapps/on_device_unit_tests/
python setup.py apk --debug \
  --sdk-dir $ANDROID_SDK_HOME \
  --ndk-dir $ANDROID_NDK_HOME \
  --requirements python3,kivy,aubio \
  --arch=armeabi-v7a --arch=arm64-v8a
```
If you have some doubts you can also take a look at the workflow file `.github/workflows/push.yml` and `Makefile` to see how we leverage docker to build recipes

Edit: jus saw we cross posted, so yeah you spotted the issue well, look at the other recipes to see how this is achieved and find a good setup to iterate on the changes, I like the Docker setup personally

jithesh82 commented:
Thank you so much.  Thanks a lot for all the notes.  I'll do it.  I believe
you read my previous comment regarding numpy errors.  I just tried this
small hack in the setup.py

`import os
os.system('pip install -U numpy')
------------lines.........
os.system('pip uninstall numpy -y')   # last line
`
and issued python setup.py build,  this time build works without 'numpy
header not' found error in an environment where numpy was not installed.
This looks like a hack to me, though.  I don't know if it is proper.

I'll check out other's recipes and do the docker test you said, to see a
proper way of doing it.

On Wed, Aug 21, 2024 at 12:25 PM Andre Miras ***@***.***>
wrote:

> OK I probably misread the code comment. I thought we meant to include
> "samplerate" in this array, but maybe we mean that having "numpy" would
> include "samplerate", I'm not sure what we mean exactly here.
>
> Also now that the "Test updated recipes" is working we see more clearly
> what the error was:
>
> python/ext/aubio-types.h:16:10: fatal error: 'numpy/arrayobject.h' file not found
> #include <numpy/arrayobject.h>
>
> Check on other recipes how numpy headers are handled.
> I would recommend trying locally from a docker environment to reproduce
> the setup we have in the CI.
>
> docker build --tag=kivy/python-for-android .
> time docker run -it --volume $(pwd)/pythonforandroid:/home/user/app/pythonforandroid --env ANDROID_NDK_HOME_LEGACY=/home/user/.android/android-ndk-legacy --env ANDROID_SDK_HOME=/home/user/.android/android-sdk --env ANDROID_NDK_HOME=/home/user/.android/android-ndk --rm kivy/python-for-android bash
>
> The volume mount is so that you can edit your recipe without always
> rebuilding the container.
> Once you're inside the running container you can try to build your apk e.g.
>
> . venv/bin/activate && cd testapps/on_device_unit_tests/ && python setup.py apk --debug --sdk-dir $ANDROID_SDK_HOME --ndk-dir $ANDROID_NDK_HOME --requirements python3,kivy,aubio --arch=armeabi-v7a --arch=arm64-v8a
>
> If you have some doubts you can also take a look at the workflow file
> .github/workflows/push.yml and Makefile to see how we leverage docker to
> build recipes
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2302496032>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLR2RGTRY6K2VUCG5HNLZSS5PPAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMBSGQ4TMMBTGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

jithesh82 committed:
Update __init__.py

AndreMiras commented on pythonforandroid/recipes/aubio/__init__.py:
```diff
@@ -0,0 +1,12 @@
+
+
+class AubioRecipe(PyProjectRecipe):
+    version = "0.4.7"
```
why not 0.4.9?

AndreMiras reviewed (commented):
Things are getting better even though it doesn't compile yet from macOS it seems, but that might be OK for this iteration.
I see you're targeting 0.4.7, why not 0.4.9 the latest version (which is already 4 years old).

AndreMiras commented:
For the macOS build issue you can try to patch the setup to remove the extra_link_args part.
https://github.com/aubio/aubio/blob/0.4.9/setup.py#L30-L32
For context the build error was:
```
/Users/runner/.android/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang -target aarch64-linux-android24 -fomit-frame-pointer -march=armv8-a -fPIC -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -L/Users/runner/.python-for-android/build/libs_collections/bdist_unit_tests_app/arm64-v8a -L/Users/runner/.python-for-android/build/other_builds/python3/arm64-v8a__ndk_target_24/python3/android-build -lpython3.11 -target aarch64-linux-android24 -fomit-frame-pointer -march=armv8-a -fPIC -I/Users/runner/.python-for-android/build/other_builds/python3/arm64-v8a__ndk_target_24/python3/Include -DANDROID -I/Users/runner/.android/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/include -I/Users/runner/.python-for-android/build/python-installs/bdist_unit_tests_app/arm64-v8a/include/python3.1 build/temp.macosx-14.6-arm64-cpython-311/python/ext/aubiomodule.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/aubioproxy.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-cvec.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-fft.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-filter.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-filterbank.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-musicutils.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-phasevoc.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-sink.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-source.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/ufuncs.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/aubio-generated.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-dct.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-mfcc.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-notes.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-onset.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-pitch.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-sampler.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-specdesc.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-tempo.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-tss.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-wavetable.o build/temp.macosx-14.6-arm64-cpython-311/src/cvec.o build/temp.macosx-14.6-arm64-cpython-311/src/fmat.o build/temp.macosx-14.6-arm64-cpython-311/src/fvec.o build/temp.macosx-14.6-arm64-cpython-311/src/io/audio_unit.o build/temp.macosx-14.6-arm64-cpython-311/src/io/ioutils.o build/temp.macosx-14.6-arm64-cpython-311/src/io/sink.o build/temp.macosx-14.6-arm64-cpython-311/src/io/sink_apple_audio.o build/temp.macosx-14.6-arm64-cpython-311/src/io/sink_sndfile.o build/temp.macosx-14.6-arm64-cpython-311/src/io/sink_wavwrite.o build/temp.macosx-14.6-arm64-cpython-311/src/io/source.o build/temp.macosx-14.6-arm64-cpython-311/src/io/source_apple_audio.o build/temp.macosx-14.6-arm64-cpython-311/src/io/source_avcodec.o build/temp.macosx-14.6-arm64-cpython-311/src/io/source_sndfile.o build/temp.macosx-14.6-arm64-cpython-311/src/io/source_wavread.o build/temp.macosx-14.6-arm64-cpython-311/src/io/utils_apple_audio.o build/temp.macosx-14.6-arm64-cpython-311/src/lvec.o build/temp.macosx-14.6-arm64-cpython-311/src/mathutils.o build/temp.macosx-14.6-arm64-cpython-311/src/notes/notes.o build/temp.macosx-14.6-arm64-cpython-311/src/onset/onset.o build/temp.macosx-14.6-arm64-cpython-311/src/onset/peakpicker.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitch.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchfcomb.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchmcomb.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchschmitt.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchspecacf.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchyin.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchyinfast.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchyinfft.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/awhitening.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct_accelerate.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct_fftw.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct_ipp.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct_ooura.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct_plain.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/fft.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/filterbank.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/filterbank_mel.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/mfcc.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/ooura_fft8g.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/phasevoc.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/specdesc.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/statistics.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/tss.o build/temp.macosx-14.6-arm64-cpython-311/src/synth/sampler.o build/temp.macosx-14.6-arm64-cpython-311/src/synth/wavetable.o build/temp.macosx-14.6-arm64-cpython-311/src/tempo/beattracking.o build/temp.macosx-14.6-arm64-cpython-311/src/tempo/tempo.o build/temp.macosx-14.6-arm64-cpython-311/src/temporal/a_weighting.o build/temp.macosx-14.6-arm64-cpython-311/src/temporal/biquad.o build/temp.macosx-14.6-arm64-cpython-311/src/temporal/c_weighting.o build/temp.macosx-14.6-arm64-cpython-311/src/temporal/filter.o build/temp.macosx-14.6-arm64-cpython-311/src/temporal/resampler.o build/temp.macosx-14.6-arm64-cpython-311/src/utils/hist.o build/temp.macosx-14.6-arm64-cpython-311/src/utils/log.o build/temp.macosx-14.6-arm64-cpython-311/src/utils/parameter.o build/temp.macosx-14.6-arm64-cpython-311/src/utils/scale.o build/temp.macosx-14.6-arm64-cpython-311/src/utils/windll.o build/temp.macosx-14.6-arm64-cpython-311/src/vecutils.o -o build/lib.macosx-14.6-arm64-cpython-311/aubio/_aubio.cpython-311-darwin.so -framework CoreFoundation -framework AudioToolbox
ld: error: unknown argument '-framework'
ld: error: unknown argument '-framework'
ld: error: cannot open CoreFoundation: No such file or directory
ld: error: cannot open AudioToolbox: No such file or directory
clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
```

jithesh82 commented:
Hi Andre,

Thanks a lot for your work. I was happy that my docker local run passed
beyond the errors last night.

1. 0.4.9 :  I didn't know about 0.4.7 when I first started using aubio.  On
their website I could only find 0.4.7 until recently.  I could start on
0.4.9
2. I'll look at the patch and macos errors.  Thanks a lot for the log and
the suggestions - it is very helpful.

I am going to do some buildozer run on my machine in the regular way with
the updated recipe.  It had happened to me that buildozer didn't make any
complaints during compilation, but when I actually import the module on my
phone, I got a module not found error.  I'll check that.

Cheers!

On Thu, Aug 22, 2024 at 5:33 AM Andre Miras ***@***.***>
wrote:

> For the macOS build issue you can try to patch the setup to remove the
> extra_link_args part.
> https://github.com/aubio/aubio/blob/0.4.9/setup.py#L30-L32
> For context the build error was:
>
> /Users/runner/.android/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang -target aarch64-linux-android24 -fomit-frame-pointer -march=armv8-a -fPIC -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -L/Users/runner/.python-for-android/build/libs_collections/bdist_unit_tests_app/arm64-v8a -L/Users/runner/.python-for-android/build/other_builds/python3/arm64-v8a__ndk_target_24/python3/android-build -lpython3.11 -target aarch64-linux-android24 -fomit-frame-pointer -march=armv8-a -fPIC -I/Users/runner/.python-for-android/build/other_builds/python3/arm64-v8a__ndk_target_24/python3/Include -DANDROID -I/Users/runner/.android/android-ndk/toolchains/llvm/prebuilt/darwin-x86_64/sysroot/usr/include -I/Users/runner/.python-for-android/build/python-installs/bdist_unit_tests_app/arm64-v8a/include/python3.1 build/temp.macosx-14.6-arm64-cpython-311/python/ext/aubiomodule.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/aubioproxy.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-cvec.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-fft.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-filter.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-filterbank.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-musicutils.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-phasevoc.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-sink.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/py-source.o build/temp.macosx-14.6-arm64-cpython-311/python/ext/ufuncs.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/aubio-generated.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-dct.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-mfcc.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-notes.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-onset.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-pitch.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-sampler.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-specdesc.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-tempo.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-tss.o build/temp.macosx-14.6-arm64-cpython-311/python/gen/gen-wavetable.o build/temp.macosx-14.6-arm64-cpython-311/src/cvec.o build/temp.macosx-14.6-arm64-cpython-311/src/fmat.o build/temp.macosx-14.6-arm64-cpython-311/src/fvec.o build/temp.macosx-14.6-arm64-cpython-311/src/io/audio_unit.o build/temp.macosx-14.6-arm64-cpython-311/src/io/ioutils.o build/temp.macosx-14.6-arm64-cpython-311/src/io/sink.o build/temp.macosx-14.6-arm64-cpython-311/src/io/sink_apple_audio.o build/temp.macosx-14.6-arm64-cpython-311/src/io/sink_sndfile.o build/temp.macosx-14.6-arm64-cpython-311/src/io/sink_wavwrite.o build/temp.macosx-14.6-arm64-cpython-311/src/io/source.o build/temp.macosx-14.6-arm64-cpython-311/src/io/source_apple_audio.o build/temp.macosx-14.6-arm64-cpython-311/src/io/source_avcodec.o build/temp.macosx-14.6-arm64-cpython-311/src/io/source_sndfile.o build/temp.macosx-14.6-arm64-cpython-311/src/io/source_wavread.o build/temp.macosx-14.6-arm64-cpython-311/src/io/utils_apple_audio.o build/temp.macosx-14.6-arm64-cpython-311/src/lvec.o build/temp.macosx-14.6-arm64-cpython-311/src/mathutils.o build/temp.macosx-14.6-arm64-cpython-311/src/notes/notes.o build/temp.macosx-14.6-arm64-cpython-311/src/onset/onset.o build/temp.macosx-14.6-arm64-cpython-311/src/onset/peakpicker.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitch.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchfcomb.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchmcomb.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchschmitt.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchspecacf.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchyin.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchyinfast.o build/temp.macosx-14.6-arm64-cpython-311/src/pitch/pitchyinfft.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/awhitening.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct_accelerate.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct_fftw.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct_ipp.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct_ooura.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/dct_plain.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/fft.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/filterbank.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/filterbank_mel.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/mfcc.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/ooura_fft8g.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/phasevoc.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/specdesc.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/statistics.o build/temp.macosx-14.6-arm64-cpython-311/src/spectral/tss.o build/temp.macosx-14.6-arm64-cpython-311/src/synth/sampler.o build/temp.macosx-14.6-arm64-cpython-311/src/synth/wavetable.o build/temp.macosx-14.6-arm64-cpython-311/src/tempo/beattracking.o build/temp.macosx-14.6-arm64-cpython-311/src/tempo/tempo.o build/temp.macosx-14.6-arm64-cpython-311/src/temporal/a_weighting.o build/temp.macosx-14.6-arm64-cpython-311/src/temporal/biquad.o build/temp.macosx-14.6-arm64-cpython-311/src/temporal/c_weighting.o build/temp.macosx-14.6-arm64-cpython-311/src/temporal/filter.o build/temp.macosx-14.6-arm64-cpython-311/src/temporal/resampler.o build/temp.macosx-14.6-arm64-cpython-311/src/utils/hist.o build/temp.macosx-14.6-arm64-cpython-311/src/utils/log.o build/temp.macosx-14.6-arm64-cpython-311/src/utils/parameter.o build/temp.macosx-14.6-arm64-cpython-311/src/utils/scale.o build/temp.macosx-14.6-arm64-cpython-311/src/utils/windll.o build/temp.macosx-14.6-arm64-cpython-311/src/vecutils.o -o build/lib.macosx-14.6-arm64-cpython-311/aubio/_aubio.cpython-311-darwin.so -framework CoreFoundation -framework AudioToolbox
> ld: error: unknown argument '-framework'
> ld: error: unknown argument '-framework'
> ld: error: cannot open CoreFoundation: No such file or directory
> ld: error: cannot open AudioToolbox: No such file or directory
> clang-14: error: linker command failed with exit code 1 (use -v to see invocation)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2304210178>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLR33JZBYSJPV5K542O3ZSWV6NAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMBUGIYTAMJXHA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

AndreMiras commented:
Indeed their website only mention the 0.4.7 release.
The other releases are in their GitHub: https://github.com/aubio/aubio/releases/tag/0.4.9

jithesh82 committed:
Update __init__.py

commented out patches, commented out os import, updated version to 0.4.9

jithesh82 committed:
Update __init__.py

jithesh82 committed:
Update __init__.py

flaked

AndreMiras commented:
Thanks for giving v0.4.9 a try, it's looking good.
Can you confirm this is running well on Android device?

For macOS, maybe simply add a class or module docstring comment to say this recipe hasn't yet be ported to compile from macOS yet.
Something like:
```
"""
Aubio recipe.
Note that this hasn't been ported to cross compile from macOS yet, the error on 0.4.9 was:
src/aubio_priv.h:95:10: fatal error: 'Accelerate/Accelerate.h' file not found
#include <Accelerate/Accelerate.h>
"""
```

jithesh82 commented:
Yes I'll add a docstring.

I built using buildozer an app to see if the aubio import and simple
print(aubio.version) would work.  It works well.

I haven't yet tested the app that the testing suite has built on docker yet
on my phone.  I can do that.  Thank you so much for all the help.

On Fri, Aug 23, 2024 at 4:42 AM Andre Miras ***@***.***>
wrote:

> Thanks for giving v0.4.9 a try, it's looking good.
> Can you confirm this is running well on Android device?
>
> For macOS, maybe simply add a class or module docstring comment to say
> this recipe hasn't yet be ported to compile from macOS yet.
> Something like:
>
> """
> Aubio recipe.
> Note that this hasn't been ported to cross compile from macOS yet, the error on 0.4.9 was:
> src/aubio_priv.h:95:10: fatal error: 'Accelerate/Accelerate.h' file not found
> #include <Accelerate/Accelerate.h>
> """
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2306601291>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLRYDS4IXUS43ONP3L53ZS3YYHAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMBWGYYDCMRZGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

jithesh82 committed:
Update __init__.py

added a doc string to indicate the errors on macos

jithesh82 committed:
Delete pythonforandroid/recipes/aubio/patches directory

jithesh82 commented:
Hello Andre,

I worked through docker.  Added pitch measurement tests.  I ran the app on
my phone.  Did adb logcat | grep python.  Imports and tests running well
without errors.

I have updated the recipe with the docstring you gave.

Thank you!

On Fri, Aug 23, 2024 at 9:41 AM Jithesh Kuyyalil ***@***.***>
wrote:

> Yes I'll add a docstring.
>
> I built using buildozer an app to see if the aubio import and simple
> print(aubio.version) would work.  It works well.
>
> I haven't yet tested the app that the testing suite has built on docker
> yet on my phone.  I can do that.  Thank you so much for all the help.
>
> On Fri, Aug 23, 2024 at 4:42 AM Andre Miras ***@***.***>
> wrote:
>
>> Thanks for giving v0.4.9 a try, it's looking good.
>> Can you confirm this is running well on Android device?
>>
>> For macOS, maybe simply add a class or module docstring comment to say
>> this recipe hasn't yet be ported to compile from macOS yet.
>> Something like:
>>
>> """
>> Aubio recipe.
>> Note that this hasn't been ported to cross compile from macOS yet, the error on 0.4.9 was:
>> src/aubio_priv.h:95:10: fatal error: 'Accelerate/Accelerate.h' file not found
>> #include <Accelerate/Accelerate.h>
>> """
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/kivy/python-for-android/pull/3044#issuecomment-2306601291>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AEICLRYDS4IXUS43ONP3L53ZS3YYHAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGMBWGYYDCMRZGE>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>
>
> --
>
>
> -Love Note,
>
> Dr. Jithesh Kuyyalil  (He/him/his)
>
> IBO @ Amway.
>
> ------------------------------
>
> Mobile 617-304-5563
>
> Web https://jitheshkuyyalil.com  Email ***@***.***
> Blog  https://joyoflife.blog  Amway Shop: https:/
> /www.amway.com/myshop/jitheshkuyyalil
> <https://www.amway.com/myshop/jitheshkuyyalil>
>
> *I BELIEVE IN YOU*
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*

AndreMiras commented:
Lovely thanks, I'll merge if the CI is still happy with the Ubuntu build

AndreMiras reviewed (approved):
Looking CI happy on Ubuntu

jithesh82 commented:
Thank you so much for all your kind help!  Wonderful!

On Sun, Aug 25, 2024 at 5:57 AM Andre Miras ***@***.***>
wrote:

> Merged #3044 <https://github.com/kivy/python-for-android/pull/3044> into
> develop.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kivy/python-for-android/pull/3044#event-14003869800>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEICLRZTXFERX5MMNEQ2ZW3ZTGS6XAVCNFSM6AAAAABLXORRLSVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJUGAYDGOBWHE4DAMA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 


-Love Note,

Dr. Jithesh Kuyyalil  (He/him/his)

IBO @ Amway.

------------------------------

Mobile 617-304-5563

Web https://jitheshkuyyalil.com  Email ***@***.***
Blog  https://joyoflife.blog  Amway Shop: https:/
/www.amway.com/myshop/jitheshkuyyalil
<https://www.amway.com/myshop/jitheshkuyyalil>

*I BELIEVE IN YOU*
